icu4c/source/test/intltest/caltest.cpp

```diff
@@ -4086,4 +4088,69 @@ void CalendarTest::TestChineseCalendarMapping() {
 
 #endif /* #if !UCONFIG_NO_FORMATTING */
 
+void CalendarTest::TestChineseCalendarMonthInSpecialYear(void) {
+    UErrorCode status = U_ZERO_ERROR;
+    GregorianCalendar gc(status);
+    ChineseCalendar cal(Locale::getRoot(), status);
+    if (failure(status, "Constructor failed")) return;
+    struct TestCase {
+      int32_t gyear;
+      int32_t gmonth;
+      int32_t gdate;
+      int32_t cmonth;
+      int32_t cdate;
+      bool cleapmonth;
+    } cases[] = {
+        // Gregorian             Chinese Calendar
+        // First some recent date
+        // From https://www.hko.gov.hk/tc/gts/time/calendar/pdf/files/2022.pdf
+        { 2022, UCAL_DECEMBER, 15, UCAL_NOVEMBER, 22, false},
```

Comment:
I think things like UCAL_NOVEMBER are confusing here, and you might be better off just using integer constants.  The Chinese calendar isn't based on the Gregorian calendar at all, so constants with the Gregorian month names just get in the way of understanding what you're testing for here.

Reply:
The problem is ICU is using 0 base instead of 1 base for month. so the 11th month will have integer 10 here, which is also very confusing.

Reply:
change to use  x-1 for month....

Reply:
The -1 thing is ugly, but better than the alternatives, I think.  But you might want to add a line to the comments explaining what you're doing there so that people understand what the "`-1`" is.